File: src/main/java/mods/betterfoliage/client/gui/NonVerboseArrayEntry.java
package mods.betterfoliage.client.gui;

import mods.betterfoliage.client.util.RenderUtils;
import net.minecraft.client.resources.I18n;
import net.minecraft.util.EnumChatFormatting;
import cpw.mods.fml.client.config.GuiConfig;
import cpw.mods.fml.client.config.GuiConfigEntries;
import cpw.mods.fml.client.config.IConfigElement;


public class NonVerboseArrayEntry extends GuiConfigEntries.ArrayEntry {

    @SuppressWarnings("unchecked")
    public NonVerboseArrayEntry(GuiConfig owningScreen, GuiConfigEntries owningEntryList, IConfigElement<?> configElement) {
        super(owningScreen, owningEntryList, configElement);
        
        RenderUtils.stripTooltipDefaultText(toolTip);
        String shortDefaults = I18n.format("betterfoliage.arrayEntryDisplay", configElement.getDefaults().length);
        toolTip.addAll(this.mc.fontRenderer.listFormattedStringToWidth(EnumChatFormatting.AQUA + I18n.format("fml.configgui.tooltip.default", shortDefaults),300));
    }

    @Override
    public void updateValueButtonText() {
        this.btnValue.displayString = I18n.format("betterfoliage.arrayEntryDisplay", currentValues.length);
    }

}

File: src/main/kotlin/mods/octarinecore/client/gui/NonVerboseArrayEntry.kt
package mods.octarinecore.client.gui

import net.minecraft.client.resources.I18n
import net.minecraft.util.text.TextFormatting.*
import net.minecraftforge.fml.client.config.GuiConfig
import net.minecraftforge.fml.client.config.GuiConfigEntries
import net.minecraftforge.fml.client.config.IConfigElement

class NonVerboseArrayEntry(
    owningScreen: GuiConfig,
    owningEntryList: GuiConfigEntries,
    configElement: IConfigElement
) : GuiConfigEntries.ArrayEntry(owningScreen, owningEntryList, configElement) {

    init {
        stripTooltipDefaultText(toolTip as MutableList<String>)
        val shortDefaults = I18n.format("${configElement.languageKey}.arrayEntry", configElement.defaults.size)
        toolTip.addAll(mc.fontRendererObj.listFormattedStringToWidth("$AQUA${I18n.format("fml.configgui.tooltip.default", shortDefaults)}", 300))
    }

    override fun updateValueButtonText() {
        btnValue.displayString = I18n.format("${configElement.languageKey}.arrayEntry", currentValues.size)
    }

}